### PR TITLE
Added rmw_logging tests

### PR DESCRIPTION
**Pull request by ahcorde**
Added rmw_logging tests to:
  - rmw_fastrtps_cpp
  - rmw_fastrtps_dynamic_cpp
  - rmw_fastrtps_shared_cpp

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rmw_fastrtps_cpp/test/test_logging.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#include <gtest/gtest.h>
+
+#include "rmw/rmw.h"
+#include "rmw/error_handling.h"
+
+TEST(TestLogging, rmw_logging)
+{
+  rmw_ret_t ret;
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_DEBUG);
```
Comment (hidmic) [inline rmw_fastrtps_cpp/test/test_logging.cpp:23]: @ahcorde consider adding a test using a bad severity. Same elsewhere.

File: rmw_fastrtps_cpp/test/test_logging.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_INFO);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_WARN);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_ERROR);
+  EXPECT_EQ(ret, RMW_RET_OK);
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_FATAL);
+  EXPECT_EQ(ret, RMW_RET_OK);
```
Comment (hidmic) [inline rmw_fastrtps_cpp/test/test_logging.cpp:32]: @ahcorde meta: given that we know we're dealing with Fast-RTPS, we can test the side effect using [`GetVerbosity()`](https://www.eprosima.com/docs/fast-rtps/1.6.0/html/classeprosima_1_1fastrtps_1_1_log.html#a32c79f1842be0b3ecd2e9f3a329b2d92).

File: rmw_fastrtps_cpp/test/test_logging.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  EXPECT_EQ(eprosima::fastrtps::Log::Kind::Error, eprosima::fastrtps::Log::GetVerbosity());
+}
+
+TEST(TestLogging, rmw_logging_bad_verbosity)
+{
+  rmw_ret_t ret = rmw_set_log_severity(static_cast<rmw_log_severity_t>(RMW_LOG_SEVERITY_FATAL + 1));
+  EXPECT_EQ(ret, RMW_RET_ERROR);
+  EXPECT_EQ(eprosima::fastrtps::Log::Kind::Error, eprosima::fastrtps::Log::GetVerbosity());
```
Comment (hidmic) [inline rmw_fastrtps_cpp/test/test_logging.cpp:46]: @ahcorde what's the expected outcome here? To keep the verbosity unchanged? To default to error?

File: rmw_fastrtps_cpp/test/test_logging.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+#include "fastrtps/log/Log.h"
+
+#include "rmw/rmw.h"
+#include "rmw/error_handling.h"
+
+TEST(TestLogging, rmw_logging)
+{
+  rmw_ret_t ret;
+  ret = rmw_set_log_severity(RMW_LOG_SEVERITY_DEBUG);
```
Comment (hidmic) [inline rmw_fastrtps_cpp/test/test_logging.cpp:25]: @ahcorde nit: collapse the last two lines. Same elsewhere.

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

**Comment by ahcorde** [comment]
Build up-to `rmw_fastrtps_cpp` `rmw_fastrtps_shared_cpp` and `rmw_fastrtps_dynamic_cpp`. Testing only these 3 packages to check the new: `test_logging`

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12416/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7383/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10126/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12329/)

**Comment by hidmic** [comment]
CI once again, including `rmw_fastrtps_dynamic_cpp` in the build:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12462/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7429/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10174/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12381/)

**Comment by hidmic** [comment]
@ahcorde is this one ready to go?

**Pull request merged**